nimbus/src/client/http_client.rs

```diff
@@ -66,6 +67,27 @@ impl SettingsClient for Client {
         let data = resp.get("data").ok_or(Error::InvalidExperimentResponse)?;
         let mut res = Vec::new();
         for exp in data.as_array().ok_or(Error::InvalidExperimentResponse)? {
+            // Validate the schema major version matches the supported version
+            let exp_schema_version = match exp.get("schemaVersion") {
+                Some(ver) => {
+                    serde_json::from_value::<String>(ver.to_owned()).unwrap_or("".to_string())
+                }
+                None => {
+                    log::trace!("Missing schemaVersion: {:#?}", exp);
+                    continue;
+                }
+            };
+            let schema_maj_version = exp_schema_version.split(".").next().unwrap_or("");
+            let schema_version: i32 = FromStr::from_str(schema_maj_version).unwrap();
+            if schema_version != SCHEMA_VERSION {
+                log::warn!(
```

Comment:
nit: seeing experiments with a higher major version number will at some point be an expected part of operation of this component, so I feel like this could be an `.info!` rather than a `.warn!`.